pkg/gameserverallocations/controller.go

```diff
@@ -168,14 +280,21 @@ func (c *Controller) creationMutationHandler(review admv1beta1.AdmissionReview)
 	}
 
 	gsa.ApplyDefaults()
-	gs, err := c.allocate(gsa)
-	if err != nil && err != ErrNoGameServerReady {
+	var gs *v1alpha1.GameServer
+	err = Retry(allocationRetry, func() error {
+		gs, err = c.allocate(gsa)
+		return err
+	})
+
+	if err != nil {
+		if err != ErrNoGameServerReady {
+			// this will trigger syncing of the cache (assuming cache might not be up to date)
```

Comment:
Re: your earlier comment about removing UnAllocated. I would probably like the functionality to be returned, just so that we're not breaking things -- but I'd probably also be happy with a TODO:; in the comments stating that it will be returning in #600 (and I'll  make sure to update the code in #600 to fix it, assuming this PR goes through first).

How do you feel about that?

Reply:
Ooh. That's an exceptionally interesting question. My biggest concern is that the calling system is aware that there was a contention, and tries again. Returning UnAllocated, might make it think that it has none left, rather than it just failed under high load.

Throwing an error would make sense, but it would be hard to know if it's an error because something horrible happened, rather than a (hopefully) rare contention issue.

Another idea might be to add another status, `Allocated`, `UnAllocated`, `Contention` ? (or something more meaningful? I can't find the right words). How does that sound?

Reply:
I like the idea of adding another status. I am OK with Contention :+1: . We can change to something else if we find a better name.